### ISSUE TITLE
Replace of arbitrary macros does not work when banner code contains pairs of square brackets

### ISSUE THREAD

**Issue by FirstImpression** [revive-adserver/revive-adserver#789]
The replace of arbitrary macros does not work when banner code contains pairs of square brackets. 

The issue is here:
https://github.com/revive-adserver/revive-adserver/blob/master/lib/max/Delivery/adRender.php#L143

The expression that currently appears in code is: 
`preg_match_all('#{(.*?)(_enc)?}#', $code, $macros);`

This replace would fail in case of the following banner code:
```
<script>
var options = { variable1: {variable1} }
</script>
```

I think more logical expression would be:
`preg_match_all('#{([a-zA-Z0-9_-]*?)(_enc)?}#', $code, $macros);`

**Comment by mbeccati** [comment]
Nicely spotted. Would you be able to submit a PR?

**Comment by FirstImpression** [comment]
Am not that familiar with Github, but I will be happy to read about creating PRs and do it, it will just take a bit more time.

**Comment by FirstImpression** [comment]
Done. [https://github.com/revive-adserver/revive-adserver/pull/790](url)

**Comment by mbeccati** [comment]
Fixed by #790

**Comment by mbeccati** [comment]
Thanks again.
